File: 5/5.cpp
#include <iostream>
#include <math.h>
using namespace std;
int isarmstrong(int n,int c){
    int tot=0;
    for(int i=0;i<c;i++){
        int rem = n%10;
        int val = pow(rem, c);
        tot+=val;
        n/=10;
    }
    return tot;
    
}
int main(){
    int num,rem,count=0;
    cout<<"Enter a number:"<<endl;
    cin>>num;
    int number=num;
    while(num>0){
        rem=num%10;
        count+=1;
        num/=10;
    }
    int check = isarmstrong(number,count);
    if(check==number){
        cout<<"Armstrong number"<<endl;
    }
    else{
        cout<<"Not a Armstrong  number"<<endl;
    }
    return 0;
}

File: 05/5.py
import math

def isArmstrong(n,c):
    tot=0
    for i in range(c):
        rem = n%10
        tot+= rem**c
        n//=10
    return tot

num = int(input("Enter a number: "))
number=num
count = 0
while(num>0):
    rem=num%10
    count+=1
    num//=10
check = isArmstrong(number, count)
if check == number:
    print("Armstrong Number")
else:
    print("Not a Armstrong Number")